rpc/src/v1/impls/light/eth.rs

```diff
@@ -533,6 +534,10 @@ impl<T: LightChainClient + 'static> Filterable for EthClient<T> {
 		self.client.block_hash(id).map(Into::into)
 	}
 
+	fn block_body(&self, _id: BlockId) -> JsonRpcResult<Option<encoded::Body>> {
+		Err(errors::light_unimplemented(None))
```

Comment:
this can be easily implemented using the `light_fetch` helper

Reply:
`light_fetch.block(...).map(|block| /* convert into encoded body somehow */`

It _will_ stall forever if the block is unknown, however. we will introduce a timeout in the future.

Reply:
Given that it may stall, should this feature be implemented for light clients only after the timeout has been introduced?

Reply:
It would be better to do it now; it will work for real blocks and only fail when people put bad data in and call it a block hash.

Reply:
I've started work to address this concern. I've got an intermediate knowledge of Rust and I've run into an issue with lifetimes upon introducing Futures to my implementation of the polling filter. I'm a bit stuck and would really appreciate it if you could take a look at my issue by clone and compiling my code.

https://github.com/adisney/parity/tree/eip758-light-block-view

If you clone my parity fork and compile the `eip758-light-block-view` branch you'll find an error regarding lifetimes.

Thanks!

Reply:
left some comments

Reply:
@rphmeier Thanks for the help. I still seem to be having the same lifetime error. I pasted the error I'm seeing into a comment on the commit [here](https://github.com/adisney/parity/commit/a0a5433ae93ec6ffbc68797324acc8defcb7f20c).